switch/switch.js

```diff
@@ -0,0 +1,144 @@
+import {getStyle, getPseudoStyle} from './style-utils.js'
+
+class Switch extends HTMLInputElement
+{
+    _isDragging = false;
+    _recentlyDragged = false;
+    _thumbsize = 0;
+    _padding = 0;
+    _bounds = {
+        lower: 0,
+        middle: 0,
+        upper: 0,
+    };
+  
+    connectedCallback()
+    {
+        // There is some racing condition, no clue as to what, this delay seems to circumvent it for now
+        setTimeout(_ => {
+            const thumbsize = getPseudoStyle(this, 'width');
+            const padding = getStyle(this, 'padding-left') + getStyle(this, 'padding-right');
+
+            this._thumbsize = thumbsize;
+            this._padding = padding;
+            this._bounds = {
+                lower: 0,
+                middle: (this.clientWidth - padding) / 4,
+                upper: this.clientWidth - thumbsize - padding,
+            };
+
+            this.addEventListener('pointerdown', this.dragInit.bind(this));
+            this.addEventListener('pointerup', this.dragEnd.bind(this));
+            this.addEventListener('click', this.preventBlubbling.bind(this));
+
+            window.addEventListener('pointerup', this.dragEnd.bind(this));
+        }, 1000);
```

Comment:
```suggestion
          const thumbsize = getPseudoStyle(this, 'width');
          const padding = getStyle(this, 'padding-left') + getStyle(this, 'padding-right');

          this._thumbsize = thumbsize;
          this._padding = padding;
          this._bounds = {
              lower: 0,
              middle: (this.clientWidth - padding) / 4,
              upper: this.clientWidth - thumbsize - padding,
          };

          this.addEventListener('pointerdown', this.dragInit.bind(this));
          this.addEventListener('pointerup', this.dragEnd.bind(this));
          this.addEventListener('click', this.preventBlubbling.bind(this));

          window.addEventListener('pointerup', this.dragEnd.bind(this));
```

tested local and there's no timeout required afaict